src/utils.rs

```diff
@@ -1,8 +1,28 @@
-use ethers_core::types::{U256, U64};
+use ethers_core::types::{U256, U64, H160};
 use serde::{de, Deserialize};
 use serde_json::Value;
 use std::str::FromStr;
 
+pub fn deserialize_h160<'de, D>(deserializer: D) -> Result<H160, D::Error>
+where
+    D: de::Deserializer<'de>,
+{
+    Ok(match Value::deserialize(deserializer)? {
+        Value::String(s) => {
+            if s.as_str() == "0x" {
+                return Ok(H160::zero());
+            }
```

Comment:
I don't think we should return `Address:zero()` here since that would be a bit ambigious in the case that we actually do send something to the zero address for whatever reason - we should probably just change the `to` field to be an `Option<Address>`, and in the case we get a response with `0x` we set the field to `None`, indicating that it was to a newly created contract. Thoughts?

Reply:
I think None is perfect! To be honest I am just learning the rust language by porting tools I have written in python to rust now. I still have some trouble implementing...

I tested this with a contract that does a transfer of msg.value to the block.coinbase and then selfdestructs. In this case the PR does work. You can add a testcase (??) by adding this as input to the transaction:
b'`\x80`@\x81\x90RA\x904\x80\x15a\x08\xfc\x02\x91`\x00\x81\x81\x81\x85\x88\x88\xf1\x93PPPP\x15\x80\x15`.W=`\x00\x80>=`\x00\xfd[P3\xff\xfe'
Here is the bytecode as well:
6080604081905241903480156108fc02916000818181858888f19350505050158015602e573d6000803e3d6000fd5b5033fffe
I am using HexString::from_string from dependency hex-string = "0.1"

If a contract is spawned that does not selfdestruct the code is responded inside the `value` field. I currently simply neglect the field in this case. Please see the last commit I did in my fork under: https://github.com/avocadochicken/ethers-flashbots/commit/3f7b7b4e67e42b4ed6e992b912ecc3dd19b47a0f
So the problem of spawning a contract might not just influance this H160 serialization.